code/modules/antagonists/changeling/powers/fakedeath.dm

```diff
@@ -19,14 +50,19 @@
 	addtimer(CALLBACK(src, .proc/ready_to_regenerate, user), LING_FAKEDEATH_TIME, TIMER_UNIQUE)
 	return TRUE
 
-/obj/effect/proc_holder/changeling/fakedeath/proc/ready_to_regenerate(mob/user)
+/datum/action/changeling/fakedeath/proc/ready_to_regenerate(mob/user)
 	if(user && user.mind)
 		var/datum/antagonist/changeling/C = user.mind.has_antag_datum(/datum/antagonist/changeling)
 		if(C && C.purchasedpowers)
 			to_chat(user, "<span class='notice'>We are ready to revive.</span>")
-			C.purchasedpowers += new /obj/effect/proc_holder/changeling/revive(null)
+			to_chat(user, "<span class='notice'>Click the Regenerative Stasis button again to revive.</span>")
+			name = "Revive"
+			desc = "We arise once more."
+			button_icon_state = "revive"

```

Comment:
the name, desc, and button_icon_state changes aren't reflected on the button itself. everything else works but this. not sure why.

Reply:
These vars are on the action datum, call UpdateButtonIcon() to copy them over to the button.